src/Controls/src/Core/TemplatedItemsList.cs

```diff
@@ -755,7 +755,7 @@ void GroupedReset()
 				// time for right now.
 				groupProxy.HeaderContent = _itemsView.CreateDefault(ListProxy.ProxiedEnumerable);
 				groupProxy.HeaderContent.BindingContext = groupProxy;
-				groupProxy.HeaderContent.SetBinding(TextCell.TextProperty, "Name");
+				groupProxy.HeaderContent.SetBinding(TextCell.TextProperty, static (TemplatedItemsList<TView, TItem> list) => list.Name);
```

Comment:
We are using BannedApiAnalyzers for a few things in this repo, do we need to also ban `SetBinding(BindableProperty,string)`? As the typed binding used here seems like it could be better for both performance and trimming/nativeaot.

Reply:
Yes, I think that would be a good thing to do, but maybe in a follow-up PR that removes the remaining usages of this API in the app (mostly Windows and Tizen specific code). In fact, I think we should ban using the `Binding` class altogether.

We should also add `[RequiresUnreferencedCode]` to the `Binding` class so that customers get trimming warnings if they use it in their apps.

Reply:
@vitek-karas we don't need it to fix all the warnings we get when we build the `dotnet new maui` app. Binding internally uses `BindingExpression` which has trimming-unsafe code so we get a warning anyway. I think we should add it so that when customers build their apps where they use `Binding`, the warnings appear in their codebase and not in the MAUI assembly.

Reply:
> I think we should add it so that when customers build their apps where they use Binding, the warnings appear in their codebase and not in the MAUI assembly.

That was exactly the reason I had in mind :-). Thanks.